flang/lib/Lower/OpenMP.cpp

```diff
@@ -2453,6 +2526,34 @@ genEnterExitDataOp(Fortran::lower::AbstractConverter &converter,
                                    deviceOperand, nowaitAttr, mapOperands);
 }
 
+mlir::omp::UpdateDataOp
+genUpdateDataOp(Fortran::lower::AbstractConverter &converter,
```

Comment:
Do you think it's possible to use the genEnterExitOp function here and add a new case for UpdateOp inside of it? The clauses seem similar to me.

Reply:
I think that would be better indeed. Done.